instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/Config.java

```diff
@@ -54,189 +43,82 @@
+   */
+  public static void internalInitializeConfig(Config config) {
+    if (INSTANCE != DEFAULT) {
+      log.warn("Config#INSTANCE was already set earlier");
+      return;
+    }
+    INSTANCE = requireNonNull(config);
   }
 
-  // Read order: Properties -> Parent
-  private Config(Properties properties, Config parent) {
-    exporterJar = properties.getProperty(EXPORTER_JAR, parent.exporterJar);
-    exporter = properties.getProperty(EXPORTER, parent.exporter);
-
-    propagators = getPropertyListValue(properties, PROPAGATORS, parent.propagators);
-
-    traceEnabled = getPropertyBooleanValue(properties, TRACE_ENABLED, parent.traceEnabled);
-    integrationsEnabled =
-        getPropertyBooleanValue(properties, INTEGRATIONS_ENABLED, parent.integrationsEnabled);
-
-    excludedClasses =
-        getPropertyListValue(properties, TRACE_CLASSES_EXCLUDE, parent.excludedClasses);
-
-    scopeDepthLimit =
-        getPropertyIntegerValue(properties, SCOPE_DEPTH_LIMIT, parent.scopeDepthLimit);
-
-    runtimeContextFieldInjection =
-        getPropertyBooleanValue(
-            properties, RUNTIME_CONTEXT_FIELD_INJECTION, parent.runtimeContextFieldInjection);
-
-    traceAnnotations = properties.getProperty(TRACE_ANNOTATIONS, parent.traceAnnotations);
-
-    traceMethods = properties.getProperty(TRACE_METHODS, parent.traceMethods);
-    traceAnnotatedMethodsExclude =
-        properties.getProperty(
-            TRACE_ANNOTATED_METHODS_EXCLUDE, parent.traceAnnotatedMethodsExclude);
-
-    traceExecutorsAll =
-        getPropertyBooleanValue(properties, TRACE_EXECUTORS_ALL, parent.traceExecutorsAll);
-    traceExecutors = getPropertyListValue(properties, TRACE_EXECUTORS, parent.traceExecutors);
-
-    sqlNormalizerEnabled =
-        getPropertyBooleanValue(properties, SQL_NORMALIZER_ENABLED, parent.sqlNormalizerEnabled);
-
-    kafkaClientPropagationEnabled =
-        getPropertyBooleanValue(
-            properties, KAFKA_CLIENT_PROPAGATION_ENABLED, parent.kafkaClientPropagationEnabled);
+  public static Config get() {
+    return INSTANCE;
+  }
 
-    hystrixTagsEnabled =
-        getBooleanSettingFromEnvironment(HYSTRIX_TAGS_ENABLED, parent.hystrixTagsEnabled);
+  public abstract Map<String, String> getAllProperties();
 
-    endpointPeerServiceMapping =
-        getPropertyMapValue(
-            properties, ENDPOINT_PEER_SERVICE_MAPPING, parent.endpointPeerServiceMapping);
+  @Nullable
+  public String getProperty(String propertyName) {
+    return getAllProperties().get(normalizePropertyName(propertyName));
+  }
 
-    log.debug("New instance: {}", this);
+  // some integrations have '-' or '_' character in their names -- this does not work well with
+  // environment variables (where we replace every non-alphanumeric character with '.'), so we're
+  // replacing those with a dot
+  public static String normalizePropertyName(String propertyName) {
+    return PROPERTY_NAME_REPLACEMENTS.matcher(propertyName.toLowerCase()).replaceAll(".");
```

Comment:
I think this is only needed for integrationNames
```suggestion
  private static String normalizeIntegrationName(String integrationName) {
    return INTEGRATION_NAME_REPLACEMENTS.matcher(integrationName.toLowerCase()).replaceAll(".");
```